File: 추천50제/구간합구하기/11660.py
import sys
input = sys.stdin.readline
n,m = map(int,input().split())
board = [list(map(int,input().split())) for _ in range(n)]

dp = [[0]*(n+1) for _ in range(n+1)]

for i in range(n):
    for j in range(n):
        dp[i+1][j+1] = (dp[i][j+1]+dp[i+1][j]-dp[i][j])+board[i][j]

for _ in range(m):
    x1,y1,x2,y2 = list(map(int,input().split()))
    print(dp[x2][y2]-(dp[x1-1][y2]+dp[x2][y1-1]-dp[x1-1][y1-1]))

File: BOJ/11660.js
let fs = require('fs');
let input = fs
  .readFileSync(process.platform == 'linux' ? './dev/stdin' : './test.txt')
  .toString()
  .trim()
  .split('\n');
let [n, m] = input.shift().split(' ').map(Number);
console.log(n, m);
const board = input.slice(0, n).map((e) => e.split(' ').map(Number));
const find = input.slice(n).map((e) => e.split(' ').map(Number));


// 1 3 5
// 3 8